### ISSUE TITLE
Set better defaults in PythonPackage

### ISSUE THREAD

**Issue by Flamefire** [easybuilders/easybuild-easyblocks#2127]
We tend to set

```
download_dep_fail = True
use_pip = True
sanity_pip_check = True
```

for all PythonPackage ECs nowadays and pip should be the default to install PythonPackages as well as the sanity check and download_dep_fail.

Hence for EasyBuild 5.0 I propose to set this as the default which is a potentially breaking change for ECs not updated but likely to work for most packages anyway.

@boegel if agreed please add this to the 5.0 milestone

**Comment by Flamefire** [comment]
Pinging in here after discovering an issue caused by not using pip: https://github.com/easybuilders/easybuild-easyblocks/issues/2389

**Comment by boegel** [comment]
This will be fixed by #3022
